### PR TITLE
Seeding and serializing programming expressions to lessons

### PR DESCRIPTION
**Pull request by dmcavoy**
Update script seeding to include LessonsProgrammingExpression entries. 

Thanks @davidsbailey for the helpful advice and example with Standards.

## Links

- spec: [Design Doc](https://docs.google.com/document/d/1dIETQq4yYRKyHwJWf7V38Y1FqR7JBVtVKzAFrQZYcKg/edit#)
- jira ticket: [PLAT-814](https://codedotorg.atlassian.net/browse/PLAT-814)

## Testing story

Manually used the import script locally to make sure the information makes it into script_json

Added tests similar to other areas of the script seeding process to check for updating lesson programming expressions as well as deleting. 

Added tests for lessons_programming_expression join table as well.

## Follow-up work

Once this and #39274 are in we can import lesson-block relationships from CB!

## PR Checklist:

- [x] Tests provide adequate coverage
- [x] Privacy and Security impacts have been assessed
- [x] Code is well-commented
- [x] New features are translatable or updates will not break translations
- [x] Relevant documentation has been added or updated
- [x] User impact is well-understood and desirable
- [x] Pull Request is labeled appropriately
- [x] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: dashboard/lib/services/script_seed.rb
Comment (dmcavoy) [inline dashboard/lib/services/script_seed.rb:83]: change lv

File: dashboard/lib/services/script_seed.rb
Comment (davidsbailey) [inline dashboard/lib/services/script_seed.rb]: this is not good because it will do too many queries. instead, load all the ProgrammingExpressions into the seed context.

File: dashboard/lib/services/script_seed.rb
Comment (davidsbailey) [inline dashboard/lib/services/script_seed.rb:201]: also load programming expressions into the seed context here:
```
seed_context.programming_expressions = ProgrammingExpression.all
```

File: dashboard/test/lib/services/script_seed_test.rb
Comment (davidsbailey) [inline dashboard/test/lib/services/script_seed_test.rb]: I would expect there to be three PEs here, because `num_programming_expressions_per_lesson` is 2

File: dashboard/test/lib/services/script_seed_test.rb
Comment (davidsbailey) [inline dashboard/test/lib/services/script_seed_test.rb]: this part is tricky. you have to create the actual PE outside of the block. see this part of my pr on lessons standards: https://github.com/code-dot-org/code-dot-org/blob/eb6ca095dab6ca52b8e70c09988faf59d5f2361f/dashboard/test/lib/services/script_seed_test.rb#L345-L347

**Review by davidsbailey — COMMENTED** [review]
Hey @dmcavoy , I worked through a similar PR for standards this week and now I am able to offer some pointers inline. Here is a link to the relevant part of my draft PR: https://github.com/code-dot-org/code-dot-org/pull/39473/files/103e6a11e5f9864ad0b0abad5f0c055493979560..eb6ca095dab6ca52b8e70c09988faf59d5f2361f

File: dashboard/test/lib/services/script_seed_test.rb
Comment (davidsbailey) [inline dashboard/test/lib/services/script_seed_test.rb]: remove this line -- only vocab and resources are prevented from being seeded in this scenario where the course version is not yet present.

**Comment by davidsbailey** [comment]
Also, at some point, I had to run the instructions at https://github.com/code-dot-org/code-dot-org/blob/staging/TESTING.md#dashboard-tests to reset my test DB in order to get unit tests to pass. I'm not sure why. The error it fixed for me was a name validation error when creating the script object.

File: dashboard/test/lib/services/script_seed_test.rb
Comment (davidsbailey) [inline dashboard/test/lib/services/script_seed_test.rb]: standards -> programming_expressions

**Review by davidsbailey — APPROVED** [review]
Fixes look good! LGTM once drone is passing. I think the comment below will help with that.

File: dashboard/test/lib/services/script_seed_test.rb
Comment (dmcavoy) [reply dashboard/test/lib/services/script_seed_test.rb]: Oops :)

**Comment by davidsbailey** [comment]
You can go ahead and merge this once it is passing -- I've merged your branch into mine and resolved the merge conflicts.

**Pull request merged**
